### PR TITLE
[CMP-2336]Compliance Operator 1.4.1 Bugs

### PR DESCRIPTION
**Pull request by GroceryBoyJr**
Version(s): Branched from main but goes to 4.12+. Compliance Operator updates apply to all supported OCP branches.

Issue: Compliance Operator 1.4.1 update release notes https://issues.redhat.com/browse/CMP-2336

Link to docs preview: https://74715--ocpdocs-pr.netlify.app/openshift-enterprise/latest/security/compliance_operator/compliance-operator-release-notes.html

QE review:
- [X] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by GroceryBoyJr** [comment]
@xiaojiey @BhargaviGudi your QE review desired please.

**Comment by rh-tokeefe** [comment]
@GroceryBoyJr overall this looks pretty good to me.

I have some fairly minor style suggestions in a few places. You did a very good job writing the bug fixes so they all have the same structure. In a few places, I think changing the language to past tense helps more clearly present the previous behavior versus the new behavior. 

Apart from that, it looks like the QA and SME review are still required before the PR can continue on in the official review process. I am still being onboarded for the peer review process, so I will also ask @bburt-rh to give this a quick looks as well.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (rh-tokeefe) [inline security/compliance_operator/compliance-operator-release-notes.adoc:22]: The link for the advisory is not presumably because the release has not gone live yet. Just noting as a reminder to go back and check the link after the release goes live.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (rh-tokeefe) [inline security/compliance_operator/compliance-operator-release-notes.adoc:37]: You did a very good job of using the same structure in the Bug Fixes section, and I have a small suggestion for your to consider. I think the line describing the previous behavior might read just a little more clearly if it makes use of past tense. For example instead of:

"Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes are no longer accessible via SSH using the core user account. With this update, nodes remain accessible via SSH using sshkey."

How about something like this:

Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes were no longer accessible via SSH using the core user account. With this update, nodes remain accessible via SSH using sshkey.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (rh-tokeefe) [inline security/compliance_operator/compliance-operator-release-notes.adoc:43]: A similar comment about tense for this entry.

Instead of "Previously, running rule `rhcos4-service-debug-shell-disabled` would show as `FAIL` even after auto-remediation is applied. With this update, rule `rhcos4-service-debug-shell-disabled` now shows `PASS` after auto-remediation is applied."

How about:

Previously, running rule `rhcos4-service-debug-shell-disabled` would show as `FAIL` even after auto-remediation was applied. With this update, rule `rhcos4-service-debug-shell-disabled` now shows `PASS` after auto-remediation is applied.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (rh-tokeefe) [inline security/compliance_operator/compliance-operator-release-notes.adoc:49]: Similar comment. Instead of:

"Previously, when a profile is removed from a completed scan shown in `ScanSettingBinding`, Compliance Operator does not remove the old scan. Afterward, when launching a new SSB using the deleted profile, Compliance Operator fails to update the result. With this update of Compliance Operator, the new SSB now shows the new compliance check result."

Maybe use "Previously, when a profile is removed from a completed scan shown in `ScanSettingBinding`, the Compliance Operator did not remove the old scan. Afterward, when launching a new SSB using the deleted profile, the Compliance Operator failed to update the result. With this update of  the Compliance Operator, the new SSB shows the compliance check result."

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc:22]: You are correct! (I should have noted that in comment 0, which is customary)

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc:37]: Agreed

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc:49]: Agreed, done!

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (bburt-rh) [inline security/compliance_operator/compliance-operator-release-notes.adoc:35]: [According to the SSG](https://redhat-documentation.github.io/supplementary-style-guide/#rhel), we shouldn't use RHEL, unless there's an exception specific to a team's usage. Is there an exception made for this feature?

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (bburt-rh) [inline security/compliance_operator/compliance-operator-release-notes.adoc:47]:
```suggestion
* Previously the `api_server_api_priority_flowschema_catch_all` rule would provide `FAIL` status on {product-title} 4.16 clusters. With this update, the `api_server_api_priority_flowschema_catch_all` rule provides `PASS` status on {product-title} 4.16 clusters. (link:https://issues.redhat.com/browse/OCPBUGS-28918[*OCPBUGS-28918*])
```
Use an attribute for OpenShift

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (bburt-rh) [inline security/compliance_operator/compliance-operator-release-notes.adoc:49]: Is the SSB acronym defined somewhere earlier? If not, I would change the first instance to `<full spelled-out name> (SSB)`

**Review by bburt-rh — APPROVED** [review]
Made a few extra edits.

Also, @rh-tokeefe - a SME review is not required -- only QE approval. It's the writer's call whether or not to have a SME review the PR.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc:35]: @bburt-rh good catch, and I worried about that. RHCOS-based containers come in "RHEL9" and "RHEL8" flavor only, and must be chosen according to that designation. We don't have a variable name for that, so I left it.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc:47]: Ooo, yes. Damb me.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc:49]: SSB is short for "ScanSettingBinding". I'll add it in parenthesis after first use.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (xiaojiey) [reply security/compliance_operator/compliance-operator-release-notes.adoc:35]: The rule should be `ocp4-kubelet-enable-protect-kernel-sysctl-file-exist` for downstream build.
And the following description is not accurate: "With this update, after auto remediation is applied, RHEL9-based RHCOS systems will show `PASS` upon the application of this rule". We removed this rule, and use  `ocp4-kubelet-enable-protect-kernel-sysctl` instead.

**Comment by xiaojiey** [comment]
One comment from my side: https://github.com/openshift/openshift-docs/pull/74715/files#r1569786047

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc]: @xiaojiey I have made the requested change, thank you very much.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (yuumasato) [inline security/compliance_operator/compliance-operator-release-notes.adoc:41]: I think this note is a little off of what was actually done.

```suggestion
* Previously, creating a `ScanSettingBinding` with profiles of different types for multiple products would bypass a restriction against multiple products types in a binding. With this update, the product validation is relaxed and now allows multiple products regardless of the of profile types in the `ScanSettingBinding`. (link:https://issues.redhat.com/browse/OCPBUGS-26229[*OCPBUGS-26229*])
```

Comment (GroceryBoyJr) [reply security/compliance_operator/compliance-operator-release-notes.adoc:41]: @yuumasato may I omit the word "relaxed" and instead simply say "With this update, the product validation now allows multiple products....."

**Review by yuumasato — APPROVED** [review]
/lgtm

**Comment by GroceryBoyJr** [comment]
QE LGTM given in direct message.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (ShaunaDiaz) [inline security/compliance_operator/compliance-operator-release-notes.adoc:35]:
```suggestion
* Previously, for RHCOS systems using RHEL9, application of the `ocp4-kubelet-enable-protect-kernel-sysctl-file-exist` rule failed. This update replaces the rule with `ocp4-kubelet-enable-protect-kernel-sysctl`. Now, after auto remediation is applied, RHEL9-based RHCOS systems show `PASS` upon the application of this rule. (link:https://issues.redhat.com/browse/OCPBUGS-13589[*OCPBUGS-13589*])
```
No future tense (and I would remove would for the sake of minimalism).

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (ShaunaDiaz) [inline security/compliance_operator/compliance-operator-release-notes.adoc:37]:
```suggestion
* Previously, after applying compliance remediations using profile `rhcos4-e8`, the nodes were no longer accessible through SSH using the core user account. With this update, nodes remain accessible through SSH using sshkey. (link:https://issues.redhat.com/browse/OCPBUGS-1833[*OCPBUGS-1833*])
```
Per https://www.ibm.com/docs/en/ibm-style?topic=word-usage#v on "via". "Through" seemed appropriate, but of course, your call for accuracy.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (ShaunaDiaz) [inline security/compliance_operator/compliance-operator-release-notes.adoc:41]:
```suggestion
* Previously, creating a `ScanSettingBinding` with profiles of different types for multiple products bypassed a restriction against multiple products types in a binding. With this update, the product validation now allows multiple products regardless of the of profile types in the `ScanSettingBinding`. (link:https://issues.redhat.com/browse/OCPBUGS-26229[*OCPBUGS-26229*])
```

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (ShaunaDiaz) [inline security/compliance_operator/compliance-operator-release-notes.adoc:43]:
```suggestion
* Previously, running the `rhcos4-service-debug-shell-disabled` rule showed as `FAIL` even after auto-remediation was applied. With this update, running the `rhcos4-service-debug-shell-disabled` rule shows `PASS` after auto-remediation is applied. (link:https://issues.redhat.com/browse/OCPBUGS-28242[*OCPBUGS-28242*])
```
There's a rule for this structure but I've lost the reference for the moment.

File: security/compliance_operator/compliance-operator-release-notes.adoc
Comment (ShaunaDiaz) [inline security/compliance_operator/compliance-operator-release-notes.adoc:47]:
```suggestion
* Previously the `api_server_api_priority_flowschema_catch_all` rule provided `FAIL` status on {product-title} 4.16 clusters. With this update, the `api_server_api_priority_flowschema_catch_all` rule provides `PASS` status on {product-title} 4.16 clusters. (link:https://issues.redhat.com/browse/OCPBUGS-28918[*OCPBUGS-28918*])
```

**Comment by ShaunaDiaz** [comment]
@GroceryBoyJr Made some nit comments for your consideration. I haven't merged because the advisory doesn't appear to be Shipped Live yet and I didn't want to break your embargo. Please let me know when ready to merge. TY!

**Pull request merged**
